lib/gnat/parser.ex

```diff
@@ -12,6 +12,7 @@ defmodule Gnat.Parser do
   end
 
   def parse(parser, "", parsed), do: {parser, Enum.reverse(parsed)}
+  def parse(parser, "PING\r\n", _), do: {parser, [{:msg, "PING", "", ""}]}
```

Comment:
Overall I found this version more readable, but this is also the first elixir project I've read; however, I wonder if exposing PING as a topic is hiding a better solution. In the protocol, PING and MSG represent the same idea, but right here we're forcing PING to be represent a topic/subject, which is not the same idea that PING represents in the protocol.